controllers/base/application_controller.go

```diff
@@ -151,7 +151,8 @@ func (r *ApplicationReconciler) validateNodes(ctx context.Context, log logr.Logg
 					r.setCondition(app, app.Status.ErrorCondition("node should have ref.group setting")...)
 					return app, ctrl.Result{}, nil
 				}
-				if *group != "chain.arcadia.kubeagi.k8s.com.cn" && *group != "agent.arcadia.kubeagi.k8s.com.cn" {
+				// Only allow chain group or agent node as the ending node
+				if *group != "chain.arcadia.kubeagi.k8s.com.cn" && (*group != "arcadia.kubeagi.k8s.com.cn" && node.Ref.Kind != "agent") {
```

Comment:
We can use the constant definition here.

https://github.com/kubeagi/arcadia/blob/cad071c3cf6a17137579fbd98fdfead345d9733b/api/app-node/chain/v1alpha1/groupversion_info.go#L28

https://github.com/kubeagi/arcadia/blob/cad071c3cf6a17137579fbd98fdfead345d9733b/api/base/v1alpha1/groupversion_info.go#L28